src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAddressOfOperation.cs

```diff
@@ -0,0 +1,131 @@
+
+        Next (Regular) Block[B2]
+            Leaving: {R1}
+}
+
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics, compilationOptions: TestOptions.UnsafeDebugDll);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void AddressOfFlow_02()
+        {
+            string source = @"
+struct S2
+{
+    unsafe void M(bool x, S2* p1, S2* p2, int* p3)
+    /*<bind>*/
+    {
+        p3 = &(x ? p1 : p2)->i;
+    }/*</bind>*/
+    public int i;
+}";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            string expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (1)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'p3')
+          Value: 
+            IParameterReferenceOperation: p3 (OperationKind.ParameterReference, Type: System.Int32*) (Syntax: 'p3')
+
+    Jump if False (Regular) to Block[B3]
+        IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'x')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'p1')
+          Value: 
+            IParameterReferenceOperation: p1 (OperationKind.ParameterReference, Type: S2*) (Syntax: 'p1')
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'p2')
+          Value: 
+            IParameterReferenceOperation: p2 (OperationKind.ParameterReference, Type: S2*) (Syntax: 'p2')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'p3 = &(x ? p1 : p2)->i;')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32*) (Syntax: 'p3 = &(x ? p1 : p2)->i')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Int32*, IsImplicit) (Syntax: 'p3')
+              Right: 
+                IAddressOfOperation (OperationKind.AddressOf, Type: System.Int32*) (Syntax: '&(x ? p1 : p2)->i')
+                  Reference: 
+                    IFieldReferenceOperation: System.Int32 S2.i (OperationKind.FieldReference, Type: System.Int32) (Syntax: '(x ? p1 : p2)->i')
+                      Instance Receiver: 
+                        IOperation:  (OperationKind.None, Type: null, IsImplicit) (Syntax: '(x ? p1 : p2)')
```

Comment:
>Type: null, [](start = 58, length = 11)

Why is the type `null` here? I would expect `S2*`

Reply:
> Why is the type null here? I would expect S2*

Anything is a fair game for OperationKind.None. It means that whatever bound node we ran into is not handled by IOperation factory yet. There are a lot of not implemented things in the factories.

---
In reply to: [188758023](https://github.com/dotnet/roslyn/pull/26869#discussion_r188758023) [](ancestors = 188758023)

Reply:
Is the expectation that when this work is done the type here will be `S2*`? Maybe my mental model is still off here but my intuition expects to see somewhat of a consistent type flow here. The insertion of `null` strikes me as off. 

---
In reply to: [188761551](https://github.com/dotnet/roslyn/pull/26869#discussion_r188761551) [](ancestors = 188761551,188758023)